builder/builder.go

```diff
@@ -720,3 +871,74 @@ func checkRPM(path string) error {
 	}
 	return nil
 }
+
+func parseUint32(s string) (uint32, error) {
+	parsed, err := strconv.ParseUint(s, 10, 32)
+	if err != nil {
+		return 0, errors.Wrapf(err, "error parsing value %q", s)
+	}
+	return uint32(parsed), nil
+}
+
+func createCompressedArchive(dst string, srcs ...string) (err error) {
```

Comment:
Any way to re-use what you wrote for fullfiles in the swupd library?

Reply:
You noticed below the reuse of ExternalWriter :-). The other bits are more specific for each use case (e.g. tradeoff whether we compress everything and write vs. streaming, what file types to support), so not worth trying to consolidate now.